### PR TITLE
[#154441774] UI for message details screen and message details info

### PR DESCRIPTION
**Pull request by scerreto**
ANDROID

schermata 2018-06-14 alle 16 31 01
schermata 2018-06-14 alle 16 31 48

IOS

schermata 2018-06-15 alle 11 19 35
schermata 2018-06-15 alle 11 27 32

### CODE REVIEW

File: ts/components/messagges/MessageDetailsComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    this.setState({ isMessageDetailsInfoVisible: !isVisible });
+  };
+  public renderDetailsInfoLink = () => {
+    if (!this.state.isMessageDetailsInfoVisible) {
+      return (
+        <View>
+          <Text link={true}>
+            {I18n.t("messageDetails.detailsLink.showLabel")}
+          </Text>
+          <Icon name="chevron-right" />
```
Comment (fpersico) [inline ts/components/messagges/MessageDetailsComponent.tsx]: You are using the work Icon. Use the one with io-icon-font.

File: ts/components/messagges/MessageDetailsComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    super(props);
+    this.state = { isMessageDetailsInfoVisible: false };
+  }
+  public handleDetailsInfoClick = (isVisible: boolean) => {
+    this.setState({ isMessageDetailsInfoVisible: !isVisible });
+  };
+  public renderDetailsInfoLink = () => {
+    if (!this.state.isMessageDetailsInfoVisible) {
+      return (
+        <View>
```
Comment (fpersico) [inline ts/components/messagges/MessageDetailsComponent.tsx]: Is this View needed?

File: ts/components/messagges/MessageDetailsComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+          <Text link={true}>
+            {I18n.t("messageDetails.detailsLink.showLabel")}
+          </Text>
+          <Icon name="chevron-right" />
+        </View>
+      );
+    } else {
+      return (
+        <View>
```
Comment (fpersico) [inline ts/components/messagges/MessageDetailsComponent.tsx]: Same here.

File: ts/components/messagges/MessageDetailsComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    return (
+      <View>
+        <TouchableOpacity
+          onPress={() =>
+            this.handleDetailsInfoClick(this.state.isMessageDetailsInfoVisible)
+          }
+        >
+          {this.renderDetailsInfoLink()}
+        </TouchableOpacity>
+        {this.state.isMessageDetailsInfoVisible ? (
```
Comment (fpersico) [inline ts/components/messagges/MessageDetailsComponent.tsx]: Use && instead of ternary operator

File: ts/navigation/MainNavigator.tsx [hunk 5]
```diff
@@ -11,11 +11,11 @@ import ROUTES from "./routes";
 import { Platform, StyleSheet, Text } from "react-native";
 import { TabBarBottom, TabNavigator } from "react-navigation";
 import I18n from "../i18n";
-import MessagesScreen from "../screens/main/MessagesScreen";
 import ProfileScreen from "../screens/main/ProfileScreen";
 import Icon from "../theme/font-icons/io-icon-font";
 import { makeFontStyleObject } from "../theme/fonts";
 import variables from "../theme/variables";
+import MessageNavigation from "./MessagesNavigation";
```
Comment (fpersico) [inline ts/navigation/MainNavigator.tsx]: Navigation -> Navigator

File: ts/components/messagges/MessageDetailsComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
+import * as React from "react";
```
Comment (fpersico) [inline ts/components/messagges/MessageDetailsComponent.tsx]: the dir name is `messagges`, need to be `messages`

File: ts/theme/variables.ts [hunk 7]
```diff
@@ -71,10 +71,15 @@ const customVariables = Object.assign(materialVariables, {
     return this.fontSizeBase * 1.75;
   },
   lineHeightBase: 24,
+
   get lineHeight1(): number {
     return this.lineHeightBase * 0.75;
   },
 
+  get lineHeight2(): number {
+    return this.lineHeightBase * (13 / 12);
```
Comment (fpersico) [inline ts/theme/variables.ts]: 13 / 12?

File: ts/theme/components/Header.ts [hunk 8]
```diff
@@ -7,7 +7,17 @@ import variables from "../variables";
 export default (): Theme => {
   return {
     "NativeBase.Left": {
-      flex: 0.2
+      "NativeBase.Button": {
```
Comment (fpersico) [inline ts/theme/components/Header.ts]: You are changing a component used in all the App. Have you checked for side effects?

File: ts/components/messagges/MessageDetailsInfoComponent.tsx [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+  public render() {
+    const { date, service, serviceOrganizationName } = this.props;
+    const localeLanguage = I18n.locale.includes("en") ? enLocale : itLocale;
+
+    return (
+      <View>
+        <View>
+          <Text bold={true}>{I18n.t("messageDetails.infoLabels.data")}</Text>
+          <Text>
+            {" "}
```
Comment (fpersico) [inline ts/components/messagges/MessageDetailsInfoComponent.tsx]: Why we need a space? Also use Template literals

File: ts/screens/messages/MessageDetailsScreen.tsx [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+type OwnProps = Readonly<{
+  navigation: NavigationScreenProp<StateParams>;
+}>;
+
+type Props = OwnProps & NavigationInjectedProps;
+
+/**
+ * This screen show the messages to the authenticated user.
```
Comment (fpersico) [inline ts/screens/messages/MessageDetailsScreen.tsx]: Not related to this file.

File: ts/components/messagges/MessageDetailsComponent.tsx (hunk 2 above)
Comment (scerreto) [reply ts/components/messagges/MessageDetailsComponent.tsx]: Without view there is this error: "JSX expressions must have one parent element."

File: ts/components/messagges/MessageDetailsComponent.tsx (hunk 3 above)
Comment (scerreto) [reply ts/components/messagges/MessageDetailsComponent.tsx]: Without view there is this error: "JSX expressions must have one parent element."

File: ts/components/messagges/MessageDetailsComponent.tsx (hunk 2 above)
Comment (fpersico) [reply ts/components/messagges/MessageDetailsComponent.tsx]: Use React.Fragment

File: ts/components/messagges/MessageDetailsComponent.tsx (hunk 1 above)
Comment (scerreto) [reply ts/components/messagges/MessageDetailsComponent.tsx]: Fixed

File: ts/components/messagges/MessageDetailsComponent.tsx (hunk 2 above)
Comment (scerreto) [reply ts/components/messagges/MessageDetailsComponent.tsx]: Fixed

File: ts/components/messagges/MessageDetailsComponent.tsx (hunk 4 above)
Comment (scerreto) [reply ts/components/messagges/MessageDetailsComponent.tsx]: Fixed

File: ts/navigation/MainNavigator.tsx (hunk 5 above)
Comment (scerreto) [reply ts/navigation/MainNavigator.tsx]: Fixed

File: ts/components/messagges/MessageDetailsComponent.tsx (hunk 6 above)
Comment (scerreto) [reply ts/components/messagges/MessageDetailsComponent.tsx]: Fixed

File: ts/theme/components/Header.ts (hunk 8 above)
Comment (scerreto) [reply ts/theme/components/Header.ts]: I have checked in all app screen, there is not side effect.

File: ts/components/messagges/MessageDetailsInfoComponent.tsx (hunk 9 above)
Comment (scerreto) [reply ts/components/messagges/MessageDetailsInfoComponent.tsx]: Fixed

File: ts/screens/messages/MessageDetailsScreen.tsx (hunk 10 above)
Comment (scerreto) [reply ts/screens/messages/MessageDetailsScreen.tsx]: Fixed

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Remember to fix the PR title by removing WIP wording when ready
  </td>
    </tr>
  </tbody>
</table>



## Affected stories

  * 🌟 [#154441774](https://www.pivotaltracker.com/story/show/154441774): Come CIT devo poter vedere i dettagli di un messaggio base

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/theme/index.ts [hunk 11]
```diff
@@ -65,6 +67,12 @@ const theme = (): Theme => {
     },
     "NativeBase.TabContainer": {
       ...tabContainerTheme()
+    },
+    "UIComponent.MessageDetailsComponent": {
+      ...messageDetailsComponent()
```
Comment (fpersico) [inline ts/theme/index.ts]: messageDetailsComponentTheme

File: ts/theme/index.ts [hunk 12]
```diff
@@ -65,6 +67,12 @@ const theme = (): Theme => {
...
+    },
+    "UIComponent.MessageDetailsInfoComponent": {
+      ...messageDetailsInfoComponent()
```
Comment (fpersico) [inline ts/theme/index.ts]: messageDetailsInfoComponentTheme

File: ts/theme/index.ts (hunk 11 above)
Comment (scerreto) [reply ts/theme/index.ts]: Fixed with next commit

File: ts/theme/index.ts (hunk 12 above)
Comment (scerreto) [reply ts/theme/index.ts]: Fixed with next commit

File: ts/components/messages/MessageComponent.tsx [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+              serviceDepartmentName,
+              serviceOrganizationName,
+              subject
+            }
+          });
+        }}
+      >
+        <Left>
+          <Text leftAlign={true} alternativeBold={true}>
+            {serviceOrganizationName + " - " + serviceDepartmentName}
```
Comment (fpersico) [inline ts/components/messages/MessageComponent.tsx]: Use interpolation

File: ts/components/messages/MessageDetailsComponent.tsx [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+import mapPropsToStyleNames from "native-base/src/utils/mapPropsToStyleNames";
+import { TouchableOpacity } from "react-native";
+import I18n from "../../i18n";
+import MessageDetailsInfoComponent from "./MessageDetailsInfoComponent";
+
+export type OwnProps = Readonly<{
+  serviceOrganizationName: string;
+  serviceDepartmentName: string;
+  service: string;
+  key: string;
```
Comment (fpersico) [inline ts/components/messages/MessageDetailsComponent.tsx]: id

File: ts/components/messages/MessageDetailsComponent.tsx [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+  date: Date;
```
Comment (fpersico) [inline ts/components/messages/MessageDetailsComponent.tsx]: created_at

File: ts/components/messages/MessageDetailsComponent.tsx [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+
+/**
+ * Implements a component that show the message details
+ */
+class MessageDetailsComponent extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = { isMessageDetailsInfoVisible: false };
+  }
+  public handleDetailsInfoClick = (isVisible: boolean) => {
```
Comment (fpersico) [inline ts/components/messages/MessageDetailsComponent.tsx]: empty line

File: ts/components/messages/MessageDetailsComponent.tsx [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+    this.setState({ isMessageDetailsInfoVisible: !isVisible });
+  };
+  public renderDetailsInfoLink = () => {
```
Comment (fpersico) [inline ts/components/messages/MessageDetailsComponent.tsx]: empty line

File: ts/components/messages/MessageDetailsComponent.tsx [hunk 18]
```diff
@@ -0,0 +1,93 @@
...
+          <Icon name="chevron-right" />
+        </React.Fragment>
+      );
+    } else {
+      return (
+        <React.Fragment>
+          <Text link={true}>
+            {I18n.t("messageDetails.detailsLink.hideLabel")}
+          </Text>
+          <Icon name="cross" />
```
Comment (fpersico) [inline ts/components/messages/MessageDetailsComponent.tsx]: Use the IconFont component

File: ts/components/messages/MessageDetailsComponent.tsx [hunk 19]
```diff
@@ -0,0 +1,93 @@
...
+          onPress={() =>
+            this.handleDetailsInfoClick(this.state.isMessageDetailsInfoVisible)
+          }
+        >
+          <View>{this.renderDetailsInfoLink()}</View>
+        </TouchableOpacity>
+        {this.state.isMessageDetailsInfoVisible && (
+          <MessageDetailsInfoComponent
+            date={date}
+            service={service}
```
Comment (fpersico) [inline ts/components/messages/MessageDetailsComponent.tsx]: if you pass the whole service why you also need serviceDepartmentName and serviceOrganizationName

File: ts/components/messages/MessageDetailsComponent.tsx [hunk 20]
```diff
@@ -0,0 +1,93 @@
...
+import { connectStyle } from "native-base-shoutem-theme";
+import mapPropsToStyleNames from "native-base/src/utils/mapPropsToStyleNames";
+import { TouchableOpacity } from "react-native";
+import I18n from "../../i18n";
+import MessageDetailsInfoComponent from "./MessageDetailsInfoComponent";
+
+export type OwnProps = Readonly<{
+  serviceOrganizationName: string;
+  serviceDepartmentName: string;
+  service: string;
```
Comment (fpersico) [inline ts/components/messages/MessageDetailsComponent.tsx]: is this the id?

File: ts/components/messages/MessageDetailsComponent.tsx [hunk 21]
```diff
@@ -0,0 +1,93 @@
...
+        <React.Fragment>
+          <Text link={true}>
+            {I18n.t("messageDetails.detailsLink.hideLabel")}
+          </Text>
+          <Icon name="cross" />
+        </React.Fragment>
+      );
+    }
+  };
+  public render() {
```
Comment (fpersico) [inline ts/components/messages/MessageDetailsComponent.tsx]: empty line

File: ts/components/messages/MessageDetailsInfoComponent.tsx [hunk 22]
```diff
@@ -0,0 +1,71 @@
+import * as React from "react";
+
+import { format } from "date-fns";
+import enLocale from "date-fns/locale/en";
+import itLocale from "date-fns/locale/it";
+import _ from "lodash";
```
Comment (fpersico) [inline ts/components/messages/MessageDetailsInfoComponent.tsx]: import only the mudule(s) you use

File: ts/components/messages/MessageDetailsInfoComponent.tsx [hunk 23]
```diff
@@ -0,0 +1,71 @@
...
+import { Text, View } from "native-base";
+import { connectStyle } from "native-base-shoutem-theme";
+import mapPropsToStyleNames from "native-base/src/utils/mapPropsToStyleNames";
+import I18n from "../../i18n";
+
+export type OwnProps = Readonly<{
+  serviceOrganizationName: string;
+  serviceDepartmentName: string;
+  service: string;
+  date: Date;
```
Comment (fpersico) [inline ts/components/messages/MessageDetailsInfoComponent.tsx]: created_at?

File: ts/components/messages/MessageDetailsInfoComponent.tsx [hunk 24]
```diff
@@ -0,0 +1,71 @@
...
+import _ from "lodash";
+import { Text, View } from "native-base";
+import { connectStyle } from "native-base-shoutem-theme";
+import mapPropsToStyleNames from "native-base/src/utils/mapPropsToStyleNames";
+import I18n from "../../i18n";
+
+export type OwnProps = Readonly<{
+  serviceOrganizationName: string;
+  serviceDepartmentName: string;
+  service: string;
```
Comment (fpersico) [inline ts/components/messages/MessageDetailsInfoComponent.tsx]: same as above

File: ts/components/messages/MessageDetailsInfoComponent.tsx [hunk 25]
```diff
@@ -0,0 +1,71 @@
...
+
+export type Props = OwnProps;
+
+/**
+ * Implements a component that show the message details info when the link is click
+ */
+class MessageDetailsInfoComponent extends React.Component<Props> {
+  public render() {
+    const {
+      date,
```
Comment (fpersico) [inline ts/components/messages/MessageDetailsInfoComponent.tsx]: again

File: ts/components/messages/MessageDetailsInfoComponent.tsx [hunk 26]
```diff
@@ -0,0 +1,71 @@
...
+      serviceDepartmentName,
+      serviceOrganizationName,
+      service
```
Comment (fpersico) [inline ts/components/messages/MessageDetailsInfoComponent.tsx]: again

File: ts/screens/messages/MessageDetailsScreen.tsx [hunk 27]
```diff
@@ -0,0 +1,86 @@
...
+  private goBack() {
+    this.props.navigation.goBack();
+  }
+
+  public render() {
+    const {
+      subject,
+      markdown,
+      serviceOrganizationName,
+      date,
```
Comment (fpersico) [inline ts/screens/messages/MessageDetailsScreen.tsx]: again

File: ts/screens/messages/MessageDetailsScreen.tsx [hunk 28]
```diff
@@ -0,0 +1,86 @@
...
+      service,
```
Comment (fpersico) [inline ts/screens/messages/MessageDetailsScreen.tsx]: again

File: ts/navigation/MainNavigator.tsx [hunk 29]
```diff
@@ -76,7 +76,7 @@ const styles = StyleSheet.create({
 const navigation = TabNavigator(
   {
     [ROUTES.MAIN_MESSAGES]: {
```
Comment (fpersico) [inline ts/navigation/MainNavigator.tsx]: you use the same constant for 2 routes

File: ts/screens/messages/MessageDetailsScreen.tsx [hunk 30]
```diff
@@ -0,0 +1,86 @@
...
+import MessageDetailsComponent from "../../components/messages/MessageDetailsComponent";
+import AppHeader from "../../components/ui/AppHeader";
+import I18n from "../../i18n";
+
+interface ParamTypeObject {
+  subject: string;
+  serviceOrganizationName: string;
+  serviceDepartmentName: string;
+  markdown: string;
+  service: string;
```
Comment (fpersico) [inline ts/screens/messages/MessageDetailsScreen.tsx]: again

File: ts/screens/messages/MessageDetailsScreen.tsx [hunk 31]
```diff
@@ -0,0 +1,86 @@
...
+  date: Date;
```
Comment (fpersico) [inline ts/screens/messages/MessageDetailsScreen.tsx]: again

File: ts/screens/messages/MessagesScreen.tsx [hunk 32]
```diff
@@ -0,0 +1,178 @@
...
+
+const styles = StyleSheet.create({
+  tabBarUnderlineStyle: {
+    width: 0
+  },
+  activeTabStyle: {
+    borderBottomWidth: 2,
+    borderBottomColor: variables.brandPrimaryLight
+  },
+  NotActiveTabStyle: {
```
Comment (fpersico) [inline ts/screens/messages/MessagesScreen.tsx]: why the uppercase "N"?

File: ts/screens/messages/MessagesScreen.tsx [hunk 33]
```diff
@@ -0,0 +1,178 @@
...
+/**
+ * This screen show the messages to the authenticated user.
+ *
+ * TODO: Just a mocked version at the moment.
+ * Going to be replaced with real content in @https://www.pivotaltracker.com/story/show/152843981
+ */
+class MessagesScreen extends React.Component<Props, never> {
+  private didFocusSubscription?: NavigationEventSubscription;
+
+  constructor(props: Props) {
```
Comment (fpersico) [inline ts/screens/messages/MessagesScreen.tsx]: useless

File: ts/screens/messages/MessagesScreen.tsx [hunk 34]
```diff
@@ -0,0 +1,178 @@
...
+  };
+
+  public getServiceName = (senderServiceId: string): string => {
+    return this.props.services.byId[senderServiceId].service_name;
+  };
+
+  public getOrganizationName = (senderServiceId: string): string => {
+    return this.props.services.byId[senderServiceId].organization_name;
+  };
+  public getDepartmentName = (senderServiceId: string): string => {
```
Comment (fpersico) [inline ts/screens/messages/MessagesScreen.tsx]: empty line

File: ts/screens/messages/MessagesScreen.tsx [hunk 35]
```diff
@@ -0,0 +1,178 @@
...
+    return this.props.services.byId[senderServiceId].department_name;
+  };
+  public renderItem = (messageDetails: IMessageDetails) => {
```
Comment (fpersico) [inline ts/screens/messages/MessagesScreen.tsx]: empty line

File: ts/screens/messages/MessagesScreen.tsx [hunk 36]
```diff
@@ -0,0 +1,178 @@
...
+    return (
+      <MessageComponent
+        key={messageDetails.item.id}
```
Comment (fpersico) [inline ts/screens/messages/MessagesScreen.tsx]: id

File: ts/screens/messages/MessagesScreen.tsx [hunk 37]
```diff
@@ -0,0 +1,178 @@
...
+        date={messageDetails.item.created_at}
```
Comment (fpersico) [inline ts/screens/messages/MessagesScreen.tsx]: createdAt

File: ts/screens/messages/MessagesScreen.tsx [hunk 38]
```diff
@@ -0,0 +1,178 @@
...
+          messageDetails.item.sender_service_id
+        )}
+        serviceDepartmentName={this.getDepartmentName(
+          messageDetails.item.sender_service_id
+        )}
+        subject={messageDetails.item.subject}
+        navigation={this.props.navigation}
+        senderServiceId={messageDetails.item.sender_service_id}
+        markdown={messageDetails.item.markdown}
+        service={this.getServiceName(messageDetails.item.sender_service_id)}
```
Comment (fpersico) [inline ts/screens/messages/MessagesScreen.tsx]: serviceName?

File: ts/components/messages/MessageDetailsInfoComponent.tsx (hunk 22 above)
Comment (scerreto) [reply ts/components/messages/MessageDetailsInfoComponent.tsx]: lodash  is used for capitalize function (row 38 & 40)

Comment (fpersico) [reply ts/components/messages/MessageDetailsInfoComponent.tsx]: i mean to load only the capitalize module and not the whole lodash

File: ts/components/messages/MessageDetailsComponent.tsx (hunk 19 above)
Comment (scerreto) [reply ts/components/messages/MessageDetailsComponent.tsx]: Service is only service type eg. Pagamento, https://projects.invisionapp.com/d/main#/console/13806972/287731268/preview

would it be more correct to call it serviceType?

File: ts/screens/messages/MessagesScreen.tsx (hunk 38 above)
Comment (scerreto) [reply ts/screens/messages/MessagesScreen.tsx]: Fixed whit next commit. Yes, this is a service name.

File: ts/screens/messages/MessageDetailsScreen.tsx [hunk 39]
```diff
@@ -0,0 +1,78 @@
...
+  markdown: string;
+  serviceName: string;
+  createdAt: Date;
+}
+
+interface ParamType {
+  readonly details: ParamTypeObject;
+}
+
+interface StateParams extends NavigationState {
```
Comment (cloudify) [inline ts/screens/messages/MessageDetailsScreen.tsx]: why we need these 3 nested types? (StateParams -> ParamType -> ParamTypeObject)

File: ts/theme/variables.ts (hunk 7 above)
Comment (cloudify) [reply ts/theme/variables.ts]: then `* 26/24` - also add the above reason as a comment so the next reader will know

File: ts/utils/__tests__/convertDateToWordDistance.test.ts [hunk 40]
```diff
@@ -11,7 +12,9 @@ describe("convertDateToWordDistance test plan", () => {
   it("should compare now date with 24 hours earlier, expected yesterday", () => {
     const nowDate = new Date();
     const testDate = addHours(nowDate, -24);
-    expect(convertDateToWordDistance(testDate)).toBe("yesterday");
+    expect(convertDateToWordDistance(testDate)).toBe(
```
Comment (cloudify) [inline ts/utils/__tests__/convertDateToWordDistance.test.ts]: what language does this test use?

`convertDateToWordDistance` shouldn't directly depend on `I18n`, instead add a parameter to `convertDateToWordDistance` that accepts a `translator` function with type `("messages.yesterday" | ... ) => string` (the caller will pass `I18n.t`) + document all the possible cases that the `translator` should cover

File: ts/screens/messages/MessageDetailsScreen.tsx (hunk 39 above)
Comment (scerreto) [reply ts/screens/messages/MessageDetailsScreen.tsx]: StateParams are the details message param passed from the message component:
details: {
              createdAt,
              markdown,
              serviceName,
              serviceDepartmentName,
              serviceOrganizationName,
              subject
}

I kept the same syntax as the pr already merged like (screens/wallet/TransactionsScreen.tsx)

File: ts/utils/__tests__/convertDateToWordDistance.test.ts (hunk 40 above)
Comment (scerreto) [reply ts/utils/__tests__/convertDateToWordDistance.test.ts]: The language test is the default language of i18n en:

// Define the supported translations
I18n.translations = {
  en,
  it
};
the purpose of the test is to verify the temporal distance, not the labels.
If we want we can exclude I18n from the test:
expect(convertDateToWordDistance(testDate)).toBe(
      "yesterday"
    );

Comment (cloudify) [reply ts/utils/__tests__/convertDateToWordDistance.test.ts]: problem is:

* if somebody changes the translation (e.g. to `Yesterday`) -> you test breaks
* if `I18N` changes the default language to something else -> your test breaks

File: ts/utils/__tests__/convertDateToWordDistance.test.ts [hunk 41]
```diff
@@ -1,30 +1,37 @@
...
+import I18n from "../../i18n";
 import { convertDateToWordDistance } from "../convertDateToWordDistance";
 
 describe("convertDateToWordDistance test plan", () => {
   it("should compare now date with 2 hours earlier, expected  hh:mm", () => {
     const nowDate = new Date();
     const testDate = addHours(nowDate, -2);
-    expect(convertDateToWordDistance(testDate)).toBe(format(testDate, "H.mm"));
+    expect(
+      convertDateToWordDistance(testDate, I18n.t("messages.yesterday"))
```
Comment (cloudify) [inline ts/utils/__tests__/convertDateToWordDistance.test.ts]: now you don't need the `I18n` dependency anymore, you can just do:

```ts
expect(convertDateToWordDistance(testDate, "yesterday")).toBe(format(testDate, "H.mm"));
```

File: ts/utils/__tests__/convertDateToWordDistance.test.ts [hunk 42]
```diff
@@ -1,30 +1,37 @@
...
+    ).toBe(format(testDate, "H.mm"));
   });
 
   it("should compare now date with 24 hours earlier, expected yesterday", () => {
     const nowDate = new Date();
     const testDate = addHours(nowDate, -24);
-    expect(convertDateToWordDistance(testDate)).toBe("yesterday");
+    expect(
```
Comment (cloudify) [inline ts/utils/__tests__/convertDateToWordDistance.test.ts]:
```ts
expect(convertDateToWordDistance(testDate, "yesterday")).toEqual("yesterday");
```

File: ts/utils/__tests__/convertDateToWordDistance.test.ts [hunk 43]
```diff
@@ -1,30 +1,37 @@
...
+      convertDateToWordDistance(testDate, I18n.t("messages.yesterday"))
+    ).toBe(I18n.t("messages.yesterday"));
   });
   // tslint:disable-next-line:no-identical-functions
   it("should compare now date with 48 hours earlier, expected DD/MM", () => {
     const nowDate = new Date();
     const testDate = addHours(nowDate, -48);
-    expect(convertDateToWordDistance(testDate)).toBe(format(testDate, "DD/MM"));
+    expect(
```
Comment (cloudify) [inline ts/utils/__tests__/convertDateToWordDistance.test.ts]:
```ts
expect(convertDateToWordDistance(testDate, "yesterday")).toBe(format(testDate, "DD/MM"));
```

File: ts/utils/__tests__/convertDateToWordDistance.test.ts [hunk 44]
```diff
@@ -1,30 +1,37 @@
...
+    ).toBe(format(testDate, "DD/MM"));
   });
 
   it("should compare now date with last year date, expected DD/MM/YY", () => {
     const nowDate = new Date();
     const testDate = addYears(nowDate, -1);
-    expect(convertDateToWordDistance(testDate)).toBe(
-      format(testDate, "DD/MM/YY")
-    );
+    expect(
```
Comment (cloudify) [inline ts/utils/__tests__/convertDateToWordDistance.test.ts]: same...

File: ts/components/messages/MessageComponent.tsx [hunk 45]
```diff
@@ -1,17 +1,22 @@
...
 import mapPropsToStyleNames from "native-base/src/utils/mapPropsToStyleNames";
-import IconFont from "../components/ui/IconFont";
-import { convertDateToWordDistance } from "../utils/convertDateToWordDistance";
+import { NavigationScreenProp, NavigationState } from "react-navigation";
+import I18n from "../../i18n";
+import ROUTES from "../../navigation/routes";
+import { convertDateToWordDistance } from "../../utils/convertDateToWordDistance";
 
 export type OwnProps = Readonly<{
+  createdAt: Date;
```
[the remark this answers is not in the document]
Comment (cloudify) [reply ts/components/messages/MessageComponent.tsx]: how comes the compiler doesn't complain?

[the remark this answers is not in the document]
Comment (fpersico) [reply ts/components/messages/MessageComponent.tsx]: I debugged this few days ago and the cause is the `/ts/@types/native-base-types.d.ts`. In this file the connectStyle function return `any` so we loose types.

[the remark this answers is not in the document]
Comment (cloudify) [reply ts/components/messages/MessageComponent.tsx]: that is very disturbing!!! we need to fix that ASAP

[the remark this answers is not in the document]
Comment (fpersico) [reply ts/components/messages/MessageComponent.tsx]: Going to fix it

[the remark this answers is not in the document]
Comment (cloudify) [reply ts/components/messages/MessageComponent.tsx]: check this PR https://github.com/teamdigitale/italia-app/pull/254

**Pull request merged**